src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs

```diff
@@ -49,6 +59,77 @@ public async Task PutJavaScript()
 
             await PutInternal(validatedAsAdmin: false);
         }
+        
+        [RavenAction("/databases/*/indexes/test", "POST", AuthorizationStatus.ValidUser, EndpointType.Write, DisableOnCpuCreditsExhaustion = true)]
+        public async Task TestIndex()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var input = await context.ReadForMemoryAsync(RequestBodyStream(), "Input");
+
+                var testIndexParameters = JsonDeserializationServer.TestIndexParameters(input);
+                var testIndexDefinition = testIndexParameters.IndexDefinition;
+                var query = testIndexParameters.Query;
+                var queryParameters = testIndexParameters.QueryParameters;
+                int maxDocumentsPerIndex = testIndexParameters.MaxDocumentsToProcess ?? 100;
+                int waitForNonStaleResultsTimeout = testIndexParameters.WaitForNonStaleResultsTimeout ?? 15;
+                
+                if (testIndexParameters.IndexDefinition is null)
+                    throw new ArgumentException("Index must have an 'IndexDefinition' field");
+
+                if (maxDocumentsPerIndex > 10_000 || maxDocumentsPerIndex < 1)
+                    throw new ArgumentException("Number of documents to process cannot be bigger than 10 000 or less than 1.");
```

Comment:
Let's use constants for min / max and use them in the message so the values won't be hard coded in the string

Reply:
Changed